server/s3_object_metadata.cc

```diff
@@ -733,6 +735,7 @@ int S3ObjectMetadata::from_json(std::string content) {
   pvid_str = newroot["PVID"].asString();
   obj_fragments = newroot["FNo"].asUInt();
   obj_parts = newroot["PRTS"].asUInt();
+  primary_obj_size = newroot["Size"].asUInt64();
```

Comment:
Check if this is a member then call newroot["Size"].asUInt64();
Cases when upgrade happens this member may not be there, which may lead to crash

Reply:
Done. Add isMember() for "FNo", as this is the key field for the new metadata format